src/internal/connector/onedrive/collections.go

```diff
@@ -480,25 +482,29 @@ func (c *Collections) UpdateCollections(
 			continue
 		}
 
+		var (
+			itemID = ptr.Val(item.GetId())
+			ictx   = clues.Add(ctx, "update_item_id", itemID)
+		)
+
 		if item.GetParentReference() == nil ||
 			item.GetParentReference().GetId() == nil ||
 			(item.GetDeleted() == nil && item.GetParentReference().GetPath() == nil) {
-			err := clues.New("no parent reference").With("item_id", *item.GetId())
-			if item.GetName() != nil {
-				err = err.With("item_name", *item.GetName())
-			}
-
-			return err
+			et.Add(clues.New("item missing parent reference").
+				WithClues(ictx).
+				With("item_id", itemID, "item_name", ptr.Val(item.GetName())))
+			continue
```

Comment:
just curious to know about this pattern- why was the continue added, when we know in the next iteration it will go
inside
 ```
if et.Err() != nil {
			break
		}
```
block and break the for loop and return from function.

Reply:
It will not necessarily break the loop.  `et.Err()` is only non-nil if we're running `failFast: true`.  Otherwise, this loop will continue even though we added an error.